index.js

```diff
@@ -128,7 +128,7 @@ AssetRewrite.prototype.rewriteAssetPath = function (string, assetPath, replaceme
   var self = this;
   return newString.replace(new RegExp('sourceMappingURL=' + escapeRegExp(assetPath)), function(wholeMatch) {
     var replaceString = replacementPath;
-    if (self.prepend && self.prepend !== '' && (!/^sourceMappingURL=(http|https|\/\/)/.test(wholeMatch))) {
+    if (self.prepend && replaceString.indexOf(this.prepend) !== 0 && (!/^sourceMappingURL=(http|https|\/\/)/.test(wholeMatch))) {
```

Comment:
I ran the test suite for this and broccoli-asset-rev and studied the possible values for `replacementPath`. It doesn't suffer the duplicate-prepend bug, like line 121. It's always passed the un-prepended string. The string is not continually replaced in a loop.

So, in the latest diff, I've undone this `indexOf` check completely!

I did delete the redundant `self.prepend` truthy check, to be consistent with the check on line 121.